cloud/amazon/ec2_eni.py

```diff
@@ -466,7 +516,7 @@ def main():
                             ],
                            required_if=([
                                ('state', 'present', ['subnet_id']),
-                               ('state', 'absent', ['eni_id']),
+                               ('state', 'absent', ['eni_id'])
```

Comment:
I would not have a default and then required when state is present
On 15 Jun 2016 22:24, "Filipe Niero Felisbino" <notifications@github.com>
wrote:

> In cloud/amazon/ec2_eni.py
> <https://github.com/ansible/ansible-modules-extras/pull/2365#discussion_r67150973>
> :
>
> > @@ -466,7 +516,7 @@ def main():
> >                              ],
> >                             required_if=([
> >                                 ('state', 'present', ['subnet_id']),
> > -                               ('state', 'absent', ['eni_id']),
> > +                               ('state', 'absent', ['eni_id'])
>
> Attached defaults to true, and may be specified without instance_id in the
> case where the user doesn't want to bother with attachment, only the ENI
> existence. In that case, attachment_status of the interface wouldn't change.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ansible/ansible-modules-extras/pull/2365/files/f1e222d5d0be182471ec823972127133c4993de0#r67150973>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHFORjT6mOYnB3c6fn17FdNvovOD6EKDks5qL-79gaJpZM4Is6-6>
> .
>